app/controllers/rhapsody_realities.js

```diff
@@ -43,15 +43,24 @@ class RhapsodyController {
     save(req, res, next) {
         let body = req.body;
 
-       return this.rhapsodyService.createRhapsody(body)
-            .then(() => {
-                req.flash('success', 'saved successfully!')
-                res.render('admin_add_rhapsody', { layout: 'admin_main' });
-            })
-            .catch((err) => {
-                req.flash('error', 'saving failed!')
-                res.render('admin_add_rhapsody', { layout: 'admin_main' });
-            });
+        // Validate Form Input
+        let error = this.rhapsodyService.ValidateRhapsodyData(req);
+        if (error) {
+            this.logger.error('Validation Error: ', error);
+            let errorMessages = error.map( item => item.msg );
+            req.flash('error', errorMessages[0]);
```

Comment:
not sure why you are doing `errorMessages[0]` instead of flashing the entire object 🤔

Reply:
displaying the entire errors makes the tab area too big, will do though.

Reply:
better for user to know things they are doing wrong e.t.c
we need to work on the styling of that tab anyways ( we could leave it for later and mostly focus on functionalities for now ) perharps we could get @ChemicalStan to assist us with certain cosmetic touches later